internal/osbuildexecutor/runner-impl-aws-ec2.go

```diff
@@ -24,6 +30,42 @@ func prepareSources(manifest []byte, store string, extraEnv []string, result boo
 	return err
 }
 
+func waitForSI(ctx context.Context, host string) bool {
+	client := http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	for {
+		resp, err := client.Get(fmt.Sprintf("%s:8001/api/v1/", host))
+		if err != nil {
+			logrus.Debugf("Waiting for secure instance continues: %v", err)
+		}
+		if resp.StatusCode != 200 {
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				logrus.Warningf("Unable to read body waiting for secure instance: %v", err)
+				continue
+			}
+			logrus.Debugf("Waiting for secure instance continues: %s", body)
+		} else {
+			return true
+		}
+
+		select {
+		case <-ctx.Done():
+			logrus.Error("Timeout waiting for secure instance to spin up")
+			return false
+		default:
+			continue
```

Comment:
Is this `default` actually needed? given that it's a `for {}` loop, that's the default, yes?

Reply:
I think it's necessary for the loop to keep calling the executor, otherwise it'll just wait on the context being done after 1 call.